lib/tasks/install_secrets.rake

```diff
@@ -0,0 +1,99 @@
+require 'active_support/core_ext/string/strip'
+require 'aws-sdk-ssm'
+
+desc <<-DESC.strip_heredoc
+  Pull the secrets for this environment and application from the AWS Parameter
+  Store and use them to write the secrets.yml
+DESC
+task :install_secrets, [] do
+  # Load the system-wide and local env vars here since the Rails application isn't loaded
+  # for this rake task
+  Dotenv.load('/etc/.env', '.env')
+
+  # Secrets live in the AWS Parameter Store under a /env_name/parameter_namespace/
+  # hierarchy.  Several environment variables are set by the AWS cloudformation scripts.
+  #
+  # This script would take the following Parameter Store values:
+  #
+  #   /qa/interactions/secret_key = 123456
+  #   /qa/interactions/redis/namespace = interactions-dev
+  #
+  # and (over)write the following to config/secrets.yml:
+  #
+  #   production:
+  #     secret_key: 123456
+  #     redis:
+  #       namespace: interactions-dev
+
+  region = get_env_var!('REGION')
+  env_name = get_env_var!('ENV_NAME')
+  namespace = get_env_var!('SECRETS_NAMESPACE')
+
+  secrets = HashWithIndifferentAccess.new
+
+  # When we get parameters from the store, we want to ignore the env_name
+  # and the parameter namespace.  Calculate how many levels there are here
+  num_ignored_key_levels = [env_name, namespace].join('/').split('/').count
+
+  client = Aws::SSM::Client.new(region: region)
+  client.get_parameters_by_path({path: "/#{env_name}/#{namespace}/",
+                                 recursive: true,
+                                 with_decryption: true}).each do |response|
+    response.parameters.each do |parameter|
+      # break out the flattened keys and ignore the env name and namespace
+      keys = parameter.name.split('/').reject(&:blank?)[num_ignored_key_levels..-1]
+      deep_populate(secrets, keys, parameter.value)
+    end
+  end
+
+  database_secrets = secrets.delete(:database)
+  write_yaml_file("config/database.yml", {
+    production: {
+      database: database_secrets[:name],
+      host: database_secrets[:url],
+      port: database_secrets[:port],
+      username: database_secrets[:username],
+      password: database_secrets[:password],
+      adapter: "postgresql",
+      encoding: "unicode",
+      pool: '<%= ENV.fetch("RAILS_MAX_THREADS") { 10 } %>'
+    }
+  })
+
+  scout_secrets = secrets.delete('scout')
+  write_yaml_file("config/scout_apm.yml", {
+    production: {
+      key: scout_secrets[:license_key],
+      name: "OpenStaxSalesforceApi (#{env_name})",
+      # crude way to disable scout by environment
+      monitor: !/noscout/.match?(env_name),
+      ignore: %w(/ping)
+    }
+  })
+
+  write_yaml_file("config/secrets.yml", {
+    production: secrets
```

Comment:
@scb6 - I'm guessing this line takes care of everything that's not database or scout (like Salesforce))?

Reply:
correct.